src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -260,37 +283,39 @@ private function evaluateFilter(string $expr, mixed $value): array
 
     private function evaluateFilterExpression(string $expr, array $context): bool
     {
-        $expr = trim($expr);
+        $expr = JsonPathUtils::normalizeWhitespace($expr);
 
-        if (str_contains($expr, '&&')) {
-            $parts = array_map('trim', explode('&&', $expr));
+        if ($this->containsOperator($expr, '&&')) {
+            $parts = preg_split('/\s*&&\s*/', $expr);
             foreach ($parts as $part) {
-                if (!$this->evaluateFilterExpression($part, $context)) {
+                if (!$this->evaluateFilterExpression(trim($part), $context)) {
                     return false;
                 }
             }
 
             return true;
         }
 
-        if (str_contains($expr, '||')) {
-            $parts = array_map('trim', explode('||', $expr));
+        if ($this->containsOperator($expr, '||')) {
```

Comment:
You could keep using `str_contains($expr, '||')` which is likely more efficient than a regex match. It won't change the result.